bundles/org.openhab.binding.ambientweather/src/main/java/org/openhab/binding/ambientweather/internal/processor/RemoteSensor.java

```diff
@@ -0,0 +1,130 @@
+        for (int sensorNumber = 1; sensorNumber <= numberOfSensors; sensorNumber++) {
+            updateSensorChannels(handler, sensorNumber, jsonData);
+        }
+    }
+
+    /*
+     * Iterate through the JSON object and update the channels for which
+     * there are remote sensor values
+     */
+    private void updateSensorChannels(AmbientWeatherStationHandler handler, int i, final @Nullable String jsonData) {
+        String sensorNumber = String.valueOf(i);
+        StringReader stringReader = new StringReader(jsonData);
+        JsonReader reader = new JsonReader(stringReader);
+        try {
+            reader.beginObject();
+            while (reader.hasNext()) {
+                String name = reader.nextName();
+                if (("temp" + sensorNumber + "f").equals(name)) {
+                    handler.updateChannel(CHGRP_REMOTE_SENSOR + sensorNumber + "#" + CH_TEMPERATURE,
+                            new QuantityType<>(reader.nextDouble(), ImperialUnits.FAHRENHEIT));
+                } else if (("humidity" + sensorNumber).equals(name)) {
+                    handler.updateChannel(CHGRP_REMOTE_SENSOR + sensorNumber + "#" + CH_HUMIDITY,
+                            new QuantityType<>(reader.nextDouble(), SmartHomeUnits.PERCENT));
+                } else if (("soiltemp" + sensorNumber).equals(name)) {
+                    handler.updateChannel(CHGRP_REMOTE_SENSOR + sensorNumber + "#" + CH_SOIL_TEMPERATURE,
+                            new QuantityType<>(reader.nextDouble(), ImperialUnits.FAHRENHEIT));
+                } else if (("soilhum" + sensorNumber).equals(name)) {
+                    double soilMoisture = reader.nextDouble();
+                    handler.updateChannel(CHGRP_REMOTE_SENSOR + sensorNumber + "#" + CH_SOIL_MOISTURE,
+                            new QuantityType<>(soilMoisture, SmartHomeUnits.PERCENT));
+                    handler.updateChannel(CHGRP_REMOTE_SENSOR + sensorNumber + "#" + CH_SOIL_MOISTURE_LEVEL,
+                            new StringType(convertSoilMoistureToString(soilMoisture)));
+                } else if (("batt" + sensorNumber).equals(name)) {
+                    handler.updateChannel(CHGRP_REMOTE_SENSOR + sensorNumber + "#" + CH_BATTERY_INDICATOR,
+                            new StringType(reader.nextString()));
+                } else {
+                    reader.skipValue();
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("IOException from JsonReader: {}", e.getMessage(), e);
+        } finally {
+            try {
+                reader.close();
+                stringReader.close();
+            } catch (IOException e) {
+                // Eat the exception
+            }
+        }
+    }
+
+    /*
+     * Convert the soil moisture to a string representation
+     */
+    private String convertSoilMoistureToString(double soilMoisture) {
+        String result = "UNKNOWN";
+        if (soilMoisture >= 0 && soilMoisture < 33.0) {
```

Comment:
Data structure

Reply:
Done. I used a TreeMap for this. It works well but seems a bit overkill for the small number of values. Maybe I didn't implement it right. Will look forward to your feedback on this.